src/main/java/io/nats/client/SubscriptionImpl.java

```diff
@@ -256,7 +272,9 @@ void setPendingMsgsMax(int max) {
             if (conn == null) {
                 throw new IllegalStateException(ERR_BAD_SUBSCRIPTION);
             }
+            this.dlvWorkerLock();
```

Comment:
Is it *guaranteed* that `setDeliveryWorker()` will never be called *between* calls to `dlvWorkerLock()` and `dlvWorkerUnlock()` so it's not possible to make a call to `dlvWorker.unlock()` that never had `lock()` called?

Unlikely, but if it ever happens, debugging it will not be fun.

Reply:
In those cases, there is no danger since we are just capturing some state, but this would have to be changed with try/finally should it become more complex for sure.